TCAT/Views/HeaderView.swift

```diff
@@ -9,51 +9,81 @@
 import UIKit
 import SnapKit
 
-protocol AddFavoritesDelegate {
+protocol HeaderViewDelegate {
     func displayFavoritesTVC()
+    func clearRecentSearches()
 }
 
+enum buttonOption {
+    case add
+    case clear
+    case none
+}
 class HeaderView: UITableViewHeaderFooterView {
     
-    var addFavoritesDelegate: AddFavoritesDelegate?
-
-    var label: UILabel = {
-        let label = UILabel()
-        label.font = .getFont(.regular, size: 14)
-        label.textColor = Colors.secondaryText
-        return label
-    }()
+    var headerViewDelegate: HeaderViewDelegate?
+    
+    static let separatorViewHeight: CGFloat = 1
 
-    var addButton: UIButton = {
-        let button = UIButton(type: .system)
-        button.setTitle("Add", for: .normal)
-        button.setTitleColor(Colors.tcatBlue, for: .normal)
-        button.addTarget(self, action: #selector(addNewFavorite), for: .touchUpInside)
-        return button
-    }()
+    var label: UILabel!
+    var button: UIButton?
 
     @objc func addNewFavorite(sender: UIButton) {
-        addFavoritesDelegate?.displayFavoritesTVC()
+        headerViewDelegate?.displayFavoritesTVC()
+    }
+    
+    @objc func clearRecentSearches(sender: UIButton) {
+        headerViewDelegate?.clearRecentSearches()
     }
 
-    func setupView(labelText: String, displayAddButton: Bool = false) {
-        label.text = labelText
-        contentView.addSubview(label)
-
-        label.snp.makeConstraints { (make) in
-            make.leading.equalToSuperview().offset(12)
-            make.bottom.equalToSuperview().offset(-6)
+    func setupView(labelText: String? = nil, buttonType: buttonOption = .none, separatorVisible: Bool = false) {
```

Comment:
What separator are you referring to?

Reply:
On the new mapView, we have a small grey separator between sections that looks different than the way we had styled it in the old version, so i added this field to account for that and made the default false so we don't to alter any of our other functions